pkg/controller/node/nodecontroller.go

```diff
@@ -782,7 +946,38 @@ func (nc *NodeController) tryUpdateNodeStatus(node *api.Node) (time.Duration, ap
 					probeTimestamp:           nc.nodeStatusMap[node.Name].probeTimestamp,
 					readyTransitionTimestamp: nc.now(),
 				}
-				return gracePeriod, observedReadyCondition, currentReadyCondition, nil
+
+				added := false
+				if nc.useTaintBasedEvictions {
+					// node is updated to unknown, add TaintNodeUnreachable to node correspondingly.
+					added, err = nc.tryAddTaintByKeyToNode(node.Name, unversioned.TaintNodeUnreachable)
```

Comment:
Actually, you can just modify `node` object by adding/removing taints, and call plain `Update` on it (instead of `UpdateStatus`) 12 lines above. I'm not 100% sure it's a safe change, because we process slightly stale Nodes here, but it should be OK.